tests/e2e/e2e.go

```diff
@@ -19,36 +19,40 @@ func generateTimeBasedName(prefix string) string {
 }
 
 func getActiveElementFromCommandOutput(command string) string {
-	result := runCmd(command + " | sed -n '1!p' | awk 'FNR==2 { print $2 }'")
+	result := runCmdShouldPass(command + " | sed -n '1!p' | awk 'FNR==2 { print $2 }'")
 	return strings.TrimSpace(result)
 }
 
-func runCmd(cmdS string) string {
+// runCmd runs a command
+// and returns the stdout, stderr and exitcode
+func runCmd(cmdS string) (string, string, int) {
 	cmd := exec.Command("/bin/sh", "-c", cmdS)
 	fmt.Fprintf(GinkgoWriter, "Running command: %s\n", cmdS)
 	session, err := gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
 
 	// wait for the command execution to complete
 	<-session.Exited
-	Expect(session.ExitCode()).To(Equal(0))
-	Expect(err).NotTo(HaveOccurred())
 
-	return string(session.Out.Contents())
+	Expect(err).NotTo(HaveOccurred())
+	return string(session.Out.Contents()), string(session.Err.Contents()), session.ExitCode()
 }
 
-// runFailCmd runs a failing command
-// and returns the stdout
-func runFailCmd(cmdS string, exitCode int) string {
-	cmd := exec.Command("/bin/sh", "-c", cmdS)
-	fmt.Fprintf(GinkgoWriter, "Running command: %s\n", cmdS)
-	session, err := gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
-
-	// wait for the command execution to complete
-	<-session.Exited
-	Expect(session.ExitCode()).To(Equal(exitCode))
-	Expect(err).NotTo(HaveOccurred())
+// runCmdShouldPass runs a command
+// and returns stdout if passes,
+// error out otherwise
+func runCmdShouldPass(cmd string) string {
```

Comment:
Looks better now. However, I'd rather keep the function names as they were before so this function should be called `runCmd`.

Reply:
> However, I'd rather keep the function names as they were before so this function should be called ```runCmd```.

Right, even I had the similar thought. But later I thought that the name itself can be made in more talkative way like ```cmdRunShouldPass```. Moreover names ```cmdRunShouldPass``` and ```cmdRunShouldFail``` looks more expressive over ```runCmd``` and ```runCmdShouldFail```